CMakeLists.txt

```diff
@@ -278,6 +272,23 @@ configure_file(${CMAKE_SOURCE_DIR}/libraries/cli11/bash-completion/completions/c
 install(FILES libraries/cli11/bash-completion/completions/leap-util DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/bash-completion/completions COMPONENT base)
 install(FILES libraries/cli11/bash-completion/completions/cleos DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/bash-completion/completions COMPONENT base)
 
+# Add the boost submodule we used to build to our install package, so headers can be found for libtester
+install(DIRECTORY "${CMAKE_SOURCE_DIR}/libraries/boost/"
+        DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/leap_boost
+        COMPONENT dev EXCLUDE_FROM_ALL
+        PATTERN ".git/*" EXCLUDE
+        PATTERN "example/*" EXCLUDE
+        PATTERN "bench/*" EXCLUDE
+        PATTERN "doc/*" EXCLUDE
+        PATTERN "testwave/*" EXCLUDE
+        PATTERN "math/test/*" EXCLUDE
+        PATTERN "json/test/*" EXCLUDE
+        PATTERN "graph/test/*" EXCLUDE
+        PATTERN "gil/test/*" EXCLUDE
+        PATTERN "geometry/test/*" EXCLUDE
+        PATTERN "beast/test/*" EXCLUDE
```

Comment:
I wonder if this could be more generic via `REGEX`.. like `REGEX "libs/*/test EXCLUDE` or something.

But the more pressing problem is just how long it's taking to build the dev package due to all these files. On CI, on main (before #1443), `make package` takes about 30 seconds, where on this branch it's over 8 minutes. As another data point I tried on my laptop and it was 21s vs 6m30s there too.

We instruct users to `make package` in our README... but for some users it's going to take longer for the package to be created than building the software! This seems like a problem.

I don't really know what to do other then disable building leap-dev.deb by default. leap-deb.deb isn't "fully supported" at the moment. For example, we don't add leap-dev.deb to the release assets, and it's existence and purpose isn't documented anywhere afaik. Instead, leap-dev.deb is used extensively by our CI tooling so that eos-system-contracts, eos-evm, etc don't need to build their own copy of leap when testing contracts.

Reply:
I'll try the simpler exclude. BTW building the package is much faster with the AntelopeIO trimmed boost repo.

Reply:
It would be even faster than the trimmed repo if we,
* install via the existing cmake targets boost exposes
* only install what we use
* for compiled libraries, only install the .a files (technically this is part of the first item I guess)

Unfortunately all that comes with its own set of tradeoffs.